### ISSUE TITLE
Unnecessary Parens when yielding JSX Part II

### ISSUE THREAD

**Issue by brainkim** [prettier/prettier#9173]
Related issues:
https://github.com/prettier/prettier/issues/7297
https://github.com/prettier/prettier/pull/7367
https://github.com/prettier/prettier/issues/8999
https://github.com/prettier/prettier/pull/9000
https://github.com/prettier/prettier/pull/9011
https://github.com/eslint/espree/issues/355
https://github.com/acornjs/acorn-jsx/issues/82
https://github.com/acornjs/acorn-jsx/pull/115

Created because the original issue is locked.
**Prettier 2.1.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0BUABAMISbwBOAFAJSHAA6UhhANnDIToQLyEAMAbiYsA7gAsc7QlRgV0cOo2YtCATxxxWAE0IAebTgBuAPlLk4FejgDUNgL56A9IdPCV9pvZAAaEBAAHXGgAZ2RQAEMKCghRAAUohDCUCNZRCLUwvwAjCgiwAGsOAGUA-JwoAHNkOQU-OABbbLhtbRaAGQiq9AjKuAAxCAoGiJhcKuQQCPQYCF8QcRgG1gB1SXgQsrA4YqScXCN9tUmwEKyQCpDLGDi8ypHkVFSrvwArEIAPACE8wpKIhpwdoVOCPZ5wN6fYoVSrsACK6Ag8DBrBeIDKFCuFEm2QizVY8wCFAqMBWOG0MHEyAAHPw-ESIFcVnkApMiXAsUZQX4AI6I+C3QLJKYhAC0UDgLRa8wocD5OFlt16DyQT1REJAVwaOBq8g1IRh8P5oNV4L8MDxZIpVKQACZzXkpDDSA0VSAOQBWeboK4AFTxyTVaKMCgAklA2rBimBiUEAIIR4owNTsFFXez2IA)
```sh
--parser babel
```

**Input:**
```jsx
function* Counter() {
  let i = 0;
  while (true) {
    yield <div>Counter {i++}</div>;
  }
}
```

**Output:**
```jsx
function* Counter() {
  let i = 0;
  while (true) {
    yield (<div>Counter {i++}</div>);
  }
}

```

**Expected behavior:**
Input unchanged.

**Comment by brodybits** [comment]
My understanding from discussion in PR #9011 is that yield of jsx with no parens would lead to 2 problems:

- Output would lead to syntax error if it is parsed by an old version of `eslint`. While I would rather see users upgrade to latest `eslint` version, this should be considered to be a breaking change.
- This could lead to problems with adding the `espree` parser, which is currently proposed in PR #9000.

**Comment by brainkim** [comment]
The thing which is increasingly annoying me tonight is that I initially [agreed with this change](https://github.com/prettier/prettier/issues/8999#issuecomment-675891211) because I thought I was helping a fellow tinkerer of generator functions and JSX deal with some tooling, but evidence now indicates the revert was motivated by a desire to unblock a PR (#9000), and that no one actually had a problem with yielding JSX elements in the first place. Currently, there are no known developers who both use the Espree parser in tooling, and write JavaScript which yields JSX expressions.

This is a disruptive change for my framework and I hope y’all can see why I’m annoyed, especially with the downplaying of my use-case in #7367 and the PR author (@fisker) marking this issue as both “low priority” and “needing discussion.”

**Comment by fisker** [comment]
I mark this as 

- "low priority"
  Because adding parens not technology breaking your code
- "needs discussion"
  Because the problem pointed by @brodybits https://github.com/prettier/prettier/issues/9173#issuecomment-689944586

**Comment by brainkim** [comment]
> Output would lead to syntax error if it is parsed by an old version of eslint.

@brodybits It’s not actually eslint which is the issue but espree. Espree is the default but I think a lot of people use eslint with the babel or typescript parsers these days.

**Comment by fisker** [comment]
That's not try our codebase use espree https://github.com/prettier/prettier/blob/master/.eslintrc.yml
